sklearn/linear_model/ridge.py

```diff
@@ -469,7 +471,13 @@ def __init__(self, alpha=1.0, fit_intercept=True, normalize=False,
         self.random_state = random_state
 
     def fit(self, X, y, sample_weight=None):
-        X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float64,
+
+        if self.solver in ['svd', 'sparse_cg', 'cholesky', 'lsqr']:
+            _dtype = [np.float64, np.float32]
+        else:
+            _dtype = np.float64
```

Comment:
@massich I forgot to comment here, it would be better to change the switch condition to support more solvers by default:

```python
if self.solver == 'lbfgs':
    # scipy lbfgs does not support float32 yet:
    # https://github.com/scipy/scipy/issues/4873
    _dtype = np.float64
else:
    # all other solvers work at both float precision levels
    _dtype = [np.float64, np.float32]
```

I have not tested the above change. Please feel free to do a PR if it works as expected.